packages/rlc-common/src/metadata/packageJson/packageCommon.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export interface PackageCommonInfoConfig {
+  name: string;
+  nameWithoutScope?: string;
+  version: string;
+  description: string;
+  moduleKind: "esm" | "cjs";
+  withTests: boolean;
+  withSamples: boolean;
+}
+
+/**
+ * Common package.json config for a package.
+ */
+export function getPackageCommonInfo(config: PackageCommonInfoConfig) {
+  const { name, version, description } = config;
+
+  return {
+    name,
+    version,
+    description,
+    engines: {
+      node: ">=18.0.0"
+    },
+    sideEffects: false,
+    autoPublish: false,
+    ...getEntryPointInformation(config)
+  };
+}
+
+export const commonPackageDependencies = {
+  tslib: "^2.6.2"
+};
+
+export function getCommonPackageDevDependencies(
+  config: PackageCommonInfoConfig
+) {
+  return {
+    "@microsoft/api-extractor": "^7.40.3",
+    "@types/node": "^18.0.0",
+    eslint: "^8.55.0",
+    prettier: "^3.2.5",
+    rimraf: "^5.0.5",
+    mkdirp: "^2.1.2",
+    typescript: "~5.3.3",
+    ...getEsmDevDependencies(config)
+  };
+}
+
+function getEsmDevDependencies({ moduleKind }: PackageCommonInfoConfig) {
+  if (moduleKind !== "esm") {
+    return {};
+  }
+  return {
+    tshy: "1.11.1"
+  };
+}
+
+function getEntryPointInformation(config: PackageCommonInfoConfig) {
+  return {
+    ...getCjsEntrypointInformation(config),
+    ...getEsmEntrypointInformation(config)
+  };
+}
+
+function getCjsEntrypointInformation({
+  name,
+  nameWithoutScope,
+  moduleKind
+}: PackageCommonInfoConfig) {
+  if (moduleKind !== "cjs") {
+    return;
+  }
+
+  const types = `./types/${nameWithoutScope ?? name}.d.ts`;
```

Comment:
is it possible that we could generateTest or generateSample in cjs ? in which case, the `${nameWithoutScope ?? name}.d.ts` is probably not in the types folder top level ? also, in the case of multi-client, it might have the subfolder?